abbiehuynh/my-crochet-corner
# Setting Up Initial React App Skeleton with Basic Routing

## PR includes:
- creating a new React app using Vite
- organized project structure of components and pages
- implementing basic routing using ‘react-router-dom’
  - configured routes for Home, Favorites, and Profile pages
- adding a navigation bar for routing between pages
- adding simple components to verify that app is running correctly
- update: added routing for all pages and organized component structure

### Testing:
- tested the connection between the server and client
  - passes the message from the server and renders through the client onto the webpage in App.jsx file.

## Testing Instructions:
To test the client, clone the repository
```
https://github.com/abbiehuynh/my-crochet-corner.git <NameNewDirectory>
```
and run the following commands in the server:
1. `npm install`
2. `npm start`

and run the following commands in the client:
1. `npm install`
2. `npm run dev`
4. Open http://localhost:5173 in your browser
- test routing to see if each link renders a div with the name of the link.
  - example: Profile link should render “Profile”
- update: test routing for user flow
![User Flow](https://raw.githubusercontent.com/abbiehuynh/my-crochet-corner/5622b9fb7f55a199a69f18b4c547e3dda2439fa8/assets/pr/userflow-diagram.png)

### Notes:
Please review the app structure and suggest any improvements.

---
Merging client branch into app-setup branch
